src/tests/roc_audio/test_profiler.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_audio/profiler.h"
+#include "roc_core/heap_allocator.h"
+
+namespace roc {
+namespace audio {
+
+namespace {
+
+struct TestFrame {
+    TestFrame(size_t size, core::nanoseconds_t time)
+        : size_(size)
+        , time_(time) {
+    }
+
+    size_t size_;
+    core::nanoseconds_t time_;
+};
+
+const core::nanoseconds_t millisecond = 1000000L;
+const core::nanoseconds_t interval = 50 * millisecond; // 5 chunks
+const int sample_rate = 5000;                          // 50 samples / chunk
+const int num_channels = 1;
+core::HeapAllocator allocator;
+
+} // namespace
+
+TEST_GROUP(profiler) {};
+
+TEST(profiler, test_moving_average) {
+    Profiler profiler(allocator, num_channels, sample_rate, interval);
+
+    TestFrame frames[9] = {
+        TestFrame(50, 50 * core::Second),      TestFrame(25, 25 * core::Second),
+        TestFrame(25, 25 * core::Second),      TestFrame(25, 25 * core::Second),
+        TestFrame(25, 25 * core::Second / 2),  TestFrame(40, 40 * core::Second),
+        TestFrame(60, 60 * core::Second / 3),  TestFrame(50, 50 * core::Second),
+        TestFrame(125, 125 * core::Second / 3)
```

Comment:
If I understand correctly, our chunks size is 10ms, and all frames in this test are larger than 10ms. So the case when the frame is smaller than the chunk is not covered. We should cover add either in this test or add a separate test.

Reply:
@gavv this does cover cases where the entire chunk is not covered. The way the algorithm you linked to works is that it is contingent on frame speed and frame samples. In the cases like 25*core::Second/2 this is covering less than one chunk. The time elapsed is only used to calculate the speed for those samples and then according to sample rate the speeds of the frames are populated into each chunk. I have the rudimentary implementation here that expanded on top of your code that you provided earlier. I'll wait till you respond to this to implement the rest of the feedback incase this is not the approach we were intending.  https://ideone.com/h5QPck

Reply:
Ah, I see now. Looks good then.

Reply:
BTW, please don't resolve conversations by yourself, so that the reviewer could go through unresolved conversations next time. (This convention is often used with github and gitlab). To mark them as done, you can leave a comment or use "reactions".